web/modules/custom/dpl_fbi/src/Plugin/Field/FieldWidget/WorkIdSearchForMaterialWidget.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\dpl_fbi\Plugin\Field\FieldWidget;
+
+use Drupal\Core\Field\FieldItemListInterface;
+use Drupal\Core\Field\WidgetBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\dpl_react_apps\Controller\DplReactAppsController;
+
+/**
+ * Defines the 'dpl_fbi_work_id_search_for_material' field widget.
+ *
+ * @FieldWidget(
+ *   id = "dpl_fbi_work_id_search_for_material",
+ *   label = @Translation("Work ID search for material"),
+ *   field_types = {"dpl_fbi_work_id"},
+ * )
+ */
+final class WorkIdSearchForMaterialWidget extends WidgetBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function formElement(FieldItemListInterface $items, $delta, array $element, array &$form, FormStateInterface $form_state): array {
+
+    $uniqId = uniqid((string) rand(), TRUE);
+    $identifier = "{$uniqId}-{$delta}";
+
+    $element['fields_container'] = [
+      '#type' => 'container',
+      '#attributes' => [
+        'class' => ['material-search__inputs-container--hidden'],
+      ],
+    ];
+
+    $element['fields_container']['value'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Field work ID'),
+      '#default_value' => $items[$delta]->value ?? '',
+      '#attributes' => [
+        'data-field-input-work-id' => $identifier,
+      ],
+    ];
+
+    $element['fields_container']['material_type'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Field material type'),
+      '#default_value' => $items[$delta]->material_type ?? '',
+      '#attributes' => [
+        'data-field-input-material-type-id' => $identifier,
+      ],
+    ];
+
+    $element['react_app_container'] = [
+      '#type' => 'container',
+      '#theme' => 'dpl_react_app',
+      '#name' => 'material-search',
+      '#data' => [
+        'class' => ['react-app-container'],
+        'previously-selected-work-id' => $items[$delta]->value ?? NULL,
+        'previously-selected-material-type' => $items[$delta]->material_type ?? NULL,
+        'unique-identifier' => $identifier,
+        'material-search-search-input-text' => $this->t('Search for material', [], ['context' => 'material search']),
```

Comment:
I would suggest context that either used proper casing e.g. `Material search` or had the module name as a prefix e.g. `DPL FBI material search` but having looked through other places our usage is all over the place so this is fine here as well.

Reply:
Changed the casing.